pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -134,6 +134,13 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1.ApiServerSour
 	return nil
 }
 
+func (r *Reconciler) FinalizeKind(ctx context.Context, source *v1.ApiServerSource) pkgreconciler.Event {
+	logging.FromContext(ctx).Info("Deleting source")
+	// Allow for eventtypes to be cleaned up
+	source.Status.CloudEventAttributes = []duckv1.CloudEventAttributes{}
+	return nil
+}
+
```

Comment:
This update is not guaranteed to be seen by the other controller (since after FinalizeKind ends finalizer is remove and with it the resource) and also I think there is (or there should be) an owner reference that should clean up event types associated with this source.